### ISSUE TITLE
Problems with access information for some TriCore operands

### ISSUE THREAD

**Issue by b1llow** [aquynh/capstone#2526]
_Originally posted by @Rot127 in https://github.com/capstone-engine/capstone/pull/2523#pullrequestreview-2395323823_
            

Need to fix all `fixme` in `tests/details/tricore.yaml`

**Comment by b1llow** [comment]
Also note that when an operand of type `TRICORE_OP_MEM` is written, the access information is wrong.

Maybe need to construct a new Operand Class in the `td` file to fix this?

**Comment by Rot127** [comment]
> Maybe need to construct a new Operand Class in the td file to fix this?

Yeah, this would be the way. Also we would need to mark all `store`/`load` instructions with `mayLoad = 1` or `mayStore = 1`. The mapping table generator will add the read/write details for `iPtr` operands. So making the registers and immediate values of type `iPtr` might be even enough.

**Comment by Rot127** [comment]
You can also add `supplement` information to the TriCore tables. Like [here](https://github.com/capstone-engine/llvm-capstone/blob/0162e010639835e25a3fe34846dd0741c4658736/llvm/utils/TableGen/PrinterCapstone.cpp#L2717-L2730).
It assumes that there is only a single memory operand (hopefully true for TriCore) and the `mayStore`, `mayLoad` attributes are set correctly. FIxing `mayLoad`, `mauStore` ins the TriCore files is way easier then changing operands. 

Then you assign this access value whenever a memory operand is created.
